### DISCUSSION TITLE
How to change MaterialDesign PopupBox PlacementMode via Binding?

### DISCUSSION THREAD

**Discussion by ojonasplima** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3146] [Q&A] [answered]
Hi!

I have a PopupBox inside a UserControl and I want, based on a condition, change the PlacementMode via a Binding on a property defined inside of the UserControl. After I set the property, the Binding appears to work but for some reason the default placement is set on the PopupBox. Someone have an idea what is the issue? FIY: If I manually set the PlacementMode, it works normally.

Here is the XAML of the Popup with the Binding:
![image](https://user-images.githubusercontent.com/86431106/227215412-c973862f-dacd-4267-873e-2f8c2c601f62.png)

Here is the Property inside my UserControl class:
![image](https://user-images.githubusercontent.com/86431106/227215868-0205d96f-af6b-444e-b9bd-b21b519d8f35.png)

Thank you for any insight.

**Comment by nicolaihenriksen** [comment, marked as answer]
@ojonasplima I would assume it is because your `SideBar.PlacementMode` property is just a normal property and thus any changes to it are not propagated to the target of the binding; effectively turning it into a one-time binding.

You can either implement `SideBar.PlacementMode` as a `DependencyProperty`, or you could have `SideBar` implement the `INotifyPropertyChanged` interface and invoke the `PropertyChanged` event in the setter of the `PlacementMode` property.

I do notice that you have scratched out some things in the screenshot, which potentially could be attributes? If you're using the [CommunityToolkit.Mvvm](https://www.nuget.org/packages/CommunityToolkit.Mvvm), then you could achieve some of that functionality simply using [attributes on a field](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty?source=recommendations).

**Reply by ojonasplima** [reply]
Hi @nicolaihenriksen! Thank you for the answer.
Implementing `INotifyPropertyChanged` didn't worked also, so I tried with `DependencyProperty` and worked!
